sharding-and-replication.md

```diff
@@ -0,0 +1,171 @@
+CREATE TABLE IF NOT EXISTS jaeger_operations AS jaeger_operations_local ENGINE = Distributed('{cluster}', default, jaeger_operations_local, rand());
+```
+
+* The `AS <table-name>` statement creates table with the same schema as the specified one. 
+* The `Distributed` engine takes as parameters cluster , database, table name and sharding key.
+
+If the distributed table is not created on all Clickhouse nodes the Jaeger query fails to get the data from the storage.
+
+### Deploy Clickhouse
+
+Deploy Clickhouse with 2 shards:
+
+```yaml
+cat <<EOF | kubectl apply -f -
+apiVersion: "clickhouse.altinity.com/v1"
+kind: "ClickHouseInstallation"
+metadata:
+  name: jaeger
+spec:
+  configuration:
+    clusters:
+      - name: cluster1
+        layout:
+          shardsCount: 2
+EOF
+```
+
+Use the following command to run `clickhouse-client` on Clickhouse nodes and create the distributed tables:
+```bash
+kubectl exec -it statefulset.apps/chi-jaeger-cluster1-0-0 -- clickhouse-client  
+```
+
+### Plugin configuration
+
+The plugin has to be configured to write and read that from the global tables:
+
+```yaml
+address: tcp://clickhouse-jaeger:9000
+spans_table: jaeger_spans
+spans_index_table: jaeger_index
+operations_table: jaeger_operations
+```
+
+## Replication
+
+Replication as the name suggest automatically replicates the data across multiple Clickhouse nodes.
+It is used to accomplish high availability, load scaling and migration/updates.
+
+The replication uses Zookeeper. Refer to the Clickhouse operator how to deploy Zookeeper.
+
+Zookeeper allows us to use `ON CLUSTER` to automatically replicate table creation on all nodes.
+Therefore the following command can be run only on a single Clickhouse node:
+
+```sql
+CREATE TABLE IF NOT EXISTS jaeger_spans_local ON CLUSTER '{cluster}'  (
+                                                                timestamp DateTime CODEC(Delta, ZSTD(1)),
+    traceID String CODEC(ZSTD(1)),
+    model String CODEC(ZSTD(3))
+    ) ENGINE ReplicatedMergeTree('/clickhouse/tables/{shard}/jaeger_spans', '{replica}')
+    PARTITION BY toDate(timestamp)
+    ORDER BY traceID
+    SETTINGS index_granularity=1024;
+
+CREATE TABLE IF NOT EXISTS jaeger_index_local ON CLUSTER '{cluster}' (
+                                                               timestamp DateTime CODEC(Delta, ZSTD(1)),
+    traceID String CODEC(ZSTD(1)),
+    service LowCardinality(String) CODEC(ZSTD(1)),
+    operation LowCardinality(String) CODEC(ZSTD(1)),
+    durationUs UInt64 CODEC(ZSTD(1)),
+    tags Array(String) CODEC(ZSTD(1)),
+    INDEX idx_tags tags TYPE bloom_filter(0.01) GRANULARITY 64,
+    INDEX idx_duration durationUs TYPE minmax GRANULARITY 1
+    ) ENGINE ReplicatedMergeTree('/clickhouse/tables/{shard}/jaeger_index', '{replica}')
+    PARTITION BY toDate(timestamp)
+    ORDER BY (service, -toUnixTimestamp(timestamp))
+    SETTINGS index_granularity=1024;
```

Comment:
It might be useful for some to put a cap on the table size with TTL, like this:
```
...
ORDER BY (service, -toUnixTimestamp(timestamp))
TTL toDate(timestamp) + INTERVAL 2 MONTH DELETE
SETTINGS ttl_only_drop_parts=1 ...;
```
`ttl_only_drop_parts` would prevent scheduling any merges to perform a ttl cleanup and would cause the part to be just dropped when the last record in it expires.

Reply:
We have open ticket for setting TTL https://github.com/pavolloffay/jaeger-clickhouse/issues/43